recipe/test-rpath.sh

```diff
@@ -1,22 +1,52 @@
 #!/bin/bash
 
+[[ ${target_platform} == "linux-64" ]] && targetsDir="targets/x86_64-linux"
+[[ ${target_platform} == "linux-ppc64le" ]] && targetsDir="targets/ppc64le-linux"
+[[ ${target_platform} == "linux-aarch64" ]] && targetsDir="targets/sbsa-linux"
+
 errors=""
 
-for bin in `find ${PREFIX}/bin -type f`; do
-    [[ `basename "$bin"` != patchelf ]] || continue
+for item in `find $PREFIX/compute-sanitizer/ -type f`; do
+    [[ "${item}" =~ "patchelf" ]] && continue
+    [[ "${item}" =~ "/docs/" ]] && continue
+    [[ "${item}" =~ "/include/" ]] && continue
+
+    filename=$(basename "${item}")
+    echo "Artifact to test: ${filename}"
+
+    pkg_info=$(conda package -w "${item}")
+    echo "\$PKG_NAME: ${PKG_NAME}"
+    echo "\$pkg_info: ${pkg_info}"
+
+    if [[ ! "$pkg_info" == *"$PKG_NAME"* ]]; then
+        echo "Not a match, skipping ${item}"
+        continue
+    fi
+
+    echo "Match found, testing ${item}"
+
+    rpath=$(patchelf --print-rpath "${item}")
+    echo "${item} rpath: ${rpath}"
 
-    rpath=$(patchelf --print-rpath $bin)
-    echo "$bin rpath: $rpath"
-    if [[ "$rpath" != "" ]]; then
-        errors+="$bin\n"
+    if [[ ${filename} =~ \.so($|\.) ]]; then
+        if [[ $rpath != "\$ORIGIN/../lib:\$ORIGIN/../${targetsDir}/lib" ]]; then # DSO lib
+            errors+="${item}\n"
+        fi
+    elif [[ ${filename} == "compute-sanitizer" || ${filename} == "TreeLauncherSubreaper" || ${filename} == "TreeLauncherTargetLdPreloadHelper" ]]; then
```

Comment:
Same question as above about generalizing for new executable additions

Reply:
Again, I am fine with it as long as we can rely on the tarball to have set the executable permissions on the supposed executable files

Reply:
Done.